File: haskell/src/Model/Links.hs
module Model.Links where

import Data.Aeson (ToJSON(..), object, (.=))
import Data.Text (Text)
import GHC.Generics (Generic)

data Links = Links
  { lnCreadlyLink                :: Text
  , lnRestaurantAroundStationLink :: Text
  , lnAdvancedSearchYoutubeLink  :: Text
  } deriving (Show, Generic)

instance ToJSON Links where
  toJSON ln = object
    [ "creadlyLink"                .= lnCreadlyLink ln
    , "restaurantAroundStationLink" .= lnRestaurantAroundStationLink ln
    , "advancedSearchYoutubeLink"  .= lnAdvancedSearchYoutubeLink ln
    ]

File: go/model/links.go
package model

// Links represents external links exposed by the API.
type Links struct {
	CreadlyLink                 string `json:"creadlyLink"`
	RestaurantAroundStationLink string `json:"restaurantAroundStationLink"`
	AdvancedSearchYoutubeLink   string `json:"advancedSearchYoutubeLink"`
}
